blog/app/controllers/posts_controller.rb

```diff
@@ -1,26 +1,25 @@
 class PostsController < ApplicationController
-  before_action :set_post, only: %i[ show edit update destroy ]
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
 
   def index
     if user_signed_in?
       @posts = current_user.posts
     else
-      @posts = Post.all
+      @posts = Post.where(:published => true)
     end
   end
 
-  def show
-  end
+  def show()end
 
   def new
     @post = current_user.posts.build
   end
 
-  def edit
-  end
+  def edit()end
 
   def create
-    @post = current_user.posts.build(post_params)
+    @post = current_user.posts.create(post_params)
```

Comment:
do you need create and save?

Reply:
No actually create also by default saves it. I will change it to build